app/src/main/java/net/gsantner/opoc/ui/FileSearchResultSelectorDialog.java

```diff
@@ -0,0 +1,247 @@
+                CustomExpandableListAdapter adapter = new CustomExpandableListAdapter(initializer._activity, filteredGroups.first, R.layout.expandable_list_group_item, groupFrom, groupTo, filteredGroups.second, android.R.layout.simple_list_item_1, childFrom, childTo);
+
+                expandableListView.setAdapter(adapter);
+            }
+
+            @Override
+            public void onTextChanged(final CharSequence arg0, final int arg1, final int arg2, final int arg3) {
+            }
+
+            @Override
+            public void beforeTextChanged(final CharSequence arg0, final int arg1, final int arg2, final int arg3) {
+            }
+        });
+
+        expandableListView.setGroupIndicator(null);
+        expandableListView.setAdapter(adapter);
+
+        expandableListView.setOnGroupClickListener((parent, view, groupPosition, id) -> {
+
+            Map<String, Object> map = (Map<String, Object>) parent.getExpandableListAdapter().getGroup(groupPosition);
+            String path = (String) map.get(Dialog.PATH_FIELD_NAME);
+            int count = (int) map.get(Dialog.COUNT_FIELD_NAME);
+
+            if (count <= 0) {
+                if (initializer._dialog != null) {
+                    initializer._dialog.dismiss();
+                }
+
+                dialogCallback.callback(path, -1);
+            }
+
+            return false;
+        });
+
+        expandableListView.setOnChildClickListener((parent, view, groupPosition, childPosition, id) -> {
+            Map<String, Object> groupMap = (Map<String, Object>) parent.getExpandableListAdapter().getGroup(groupPosition);
+            Map<String, Object> childMap = (Map<String, Object>) parent.getExpandableListAdapter().getChild(groupPosition, childPosition);
+            String path = (String) groupMap.get(Dialog.PATH_FIELD_NAME);
+            int lineNumber = (int) childMap.get(Dialog.LINE_NUMBER_FIELD_NAME);
+            if (lineNumber >= 0) {
+                if (initializer._dialog != null) {
+                    initializer._dialog.dismiss();
+                }
+
+                dialogCallback.callback(path, lineNumber);
+            }
+
+            return false;
+        });
+
+        dialogLayout.addView(expandableListView);
+
+
+        // Configure dialog
+        dialogBuilder.setView(dialogLayout)
+                .setTitle(R.string.select)
+                .setOnCancelListener(null)
+                .setNegativeButton(android.R.string.cancel, (dialogInterface, i) -> dialogInterface.dismiss());
+
+        if (searchResults.isEmpty()) {
+            dialogBuilder.setMessage("     ¯\\_(ツ)_/¯     ");
+        }
+
+        return dialogBuilder;
+    }
+
+
+    private static Pair<ArrayList<Map<String, Object>>, ArrayList<ArrayList<Map<String, Object>>>> filter(final List<SearchEngine.FitFile> searchResults, String query) {
```

Comment:
@adelobosko If you are able to document this data structure, that would be helpful for future maintainece.